src/pybind/mgr/prometheus/module.py

```diff
@@ -1060,7 +1060,28 @@ def index(self):
             @cherrypy.expose
             def metrics(self):
                 cherrypy.response.headers['Content-Type'] = 'text/plain'
-                return ''
+                active_uri = module.get_active_uri()
+                if active_uri:
+                    module.log.info("Redirecting to active '%s'", active_uri)
+                    raise cherrypy.HTTPRedirect(active_uri + "metrics")
+                else:
+                    template = """
+                <html>
```

Comment:
“/metrics" endpoint is visited by `promethues server`. It's not a good idea to return such a web page. maybe we should return a 404 code (what does promethues want when exporter failed?)

Reply:
The `prometheus` doc suggests returning a 5xx error : `https://prometheus.io/docs/instrumenting/writing_exporters/#failed-scrapes`